### PR TITLE
added_config_type

### PR DESCRIPTION
**Pull request by peter1000**
Using Type for potential increase in configuration settings.

implements: https://github.com/MichaelHatherly/Lexicon.jl/issues/52
solves: https://github.com/MichaelHatherly/Lexicon.jl/issues/53

### CODE REVIEW

**Comment by peter1000** [comment]
Probably it's best to wait for the couple of changes we have planned before releasing a tagged version.

File: src/render/md.jl [hunk 1]
```diff
@@ -1,43 +1,30 @@
...
-    vaild_tags = vcat(HTAGS,STYLETAGS)
-    for (k, v) in mdstyle
-        k in reg_keys || error("Invalid mdstyle key:  `$k`. Valid keys: [$(join(reg_keys, ", "))].")
-        v in vaild_tags || error("Invalid mdstyle value:  `$v`. Valid values: [$(join(vaild_tags, ", "))].")
+function save(file::String, mime::MIME"text/md", doc::Metadata, config::Config)
+    # validate mdstyle
+    for k in [:mdstyle_header, :mdstyle_objname, :mdstyle_meta, :mdstyle_exported, :mdstyle_internal]
+       getfield(config, k) in vcat(HTAGS,STYLETAGS) ||
+                            error("""Invalid mdstyle value: config-item `$k -> $(config[k])`.
+                                    Valid values: [$(join(vcat(HTAGS,STYLETAGS), ", "))].""")
```
Comment (MichaelHatherly) [inline src/render/md.jl]: Can this validation code run during construction of the `Config` object to keep the errors localised to there?

**Comment by MichaelHatherly** [comment]
Other than the one inline comment all looks good.

File: src/render/md.jl (hunk 1 above)
Comment (peter1000) [inline src/render/md.jl]: I was thinking the same but thought maybe as it only related to markdown I move it there. Anyway I will change it as you suggested.

**Comment by peter1000** [comment]
I was thinking the same but thought maybe as it only related to markdown I move it there. Anyway I will change it as you suggested.

**Comment by MichaelHatherly** [comment]
AppVeyor is hanging on the 64bit build during cloning, unrelated to this PR. Since everything else has passed we'll merge now anyway.

**Pull request merged**